### ISSUE TITLE
chr prefix

### ISSUE THREAD

**Issue by hidehitofukushima** [Wedge-lab/battenberg#127]
Hi.
I'm using Battenberg recently and facing a little situation I seriously can't cope with, so I need your help.

I used the docker image from quay.io/wtsicgp/cgpbattenberg:3.7.1 , pulled it in a singularity image(singuarity pull docker://quay.io/wtsicgp/cgpbattenberg:3.7.1) and tried this command, and got these error.

####################################################################################
#$ -S /bin/bash
#$ -cwd
#$ -pe def_slot 16
#$ -l s_vmem=32G

singularity exec -e --bind /share/xxx/:/share/xxx/ /home/xxxx/database/images/cgpbattenberg_3.7.1.sif
battenberg.pl
-o ${OUTPUT_DIR}
-r ${FASTA}.fai
-tb ${TUMOR_BAM}
-nb ${NORMAL_BAM}
-e /home/xxxx/BATTENBERG/tmp2/reference_hg38/impute_info.txt
-u /home/xxxx/BATTENBERG/tmp2/reference_hg38/1000G_loci_hg38/
-ig /home/xxxx/BATTENBERG/tmp2/reference_hg38/ignore_contigs.wtchr
-gc /home/xxxx/BATTENBERG/tmp2/reference_hg38/GC_correction_hg38/
-t 16
-c /home/xxxx/BATTENBERG/tmp2/reference_hg38/probloci/probloci.txt.gz
-pr WGS
-ge XY
-ra 38

xxxxxxxxxxxxxxxx_allelecount.1.err:

/opt/wtsi-cgp/bin/alleleCounter -l /home/xxxxx/BATTENBERG/result/sample2/xxxxxxxxxxx/tmpBattenberg/1000genomesloci2012_chr1_split1.txt -b /home/xxxxxx/database/links/xxxxxx/result/wgs/xxxxxxxxxxxxxx/bam/yyyyyyyyyyyyyyyyyyyy/yyyyyyyyyyyyyyyyyyyyyyy.markdup.bam -o /home/xxxx/BATTENBERG/result/sample2/xxxxxxxxxxxxxxxxxx/tmpBattenberg/yyyyyyyyyyyyyyyyyyyyyyyyy_alleleFrequencies_chr1_split1.txt -m 20 -r /home/xxxx/database/reference/Homo_sapiens_assembly38.fasta.fai -d
Reading locis
Done reading locis
Multi pos start:
1054.64user 14.98system 17:53.85elapsed 99%CPU (0avgtext+0avgdata 4563196maxresident)k
521476inputs+119296outputs (9major+1137847minor)pagefaults 0swaps

xxxxxxxxxxxxxxxx_runbaflog.0.err:

cd /home/xxxx/some_directory/tmpBattenberg
/usr/bin/Rscript -e 'library(Battenberg); getBAFsAndLogRs(tumourAlleleCountsFile.prefix="87336744-255b-42a1-a1ff-0eded2655d8f_alleleFrequencies_chr", normalAlleleCountsFile.prefix="dace3db7-74a8-408c-9872-d424ee7429bb_alleleFrequencies_chr", figuresFile.prefix="87336744-255b-42a1-a1ff-0eded2655d8f_", BAFnormalFile="87336744-255b-42a1-a1ff-0eded2655d8f_normalBAF.tab", BAFmutantFile="87336744-255b-42a1-a1ff-0eded2655d8f_mutantBAF.tab", logRnormalFile="87336744-255b-42a1-a1ff-0eded2655d8f_normalLogR.tab", logRmutantFile="87336744-255b-42a1-a1ff-0eded2655d8f_mutantLogR.tab", combinedAlleleCountsFile="87336744-255b-42a1-a1ff-0eded2655d8f_alleleCounts.tab", chr_names=as.vector(c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X")), g1000file.prefix="/home/xxxx/BATTENBERG/tmp2/reference_hg38/1000G_loci_hg38/1000genomesAlleles2012_chr", minCounts=10, samplename="87336744-255b-42a1-a1ff-0eded2655d8f", seed=1488823153) '
511.57user 12.36system 10:01.43elapsed 87%CPU (0avgtext+0avgdata 19791232maxresident)k
785890inputs+5320592outputs (122major+2178229minor)pagefaults 0swaps

xxxxxxxxxxxxxxxx_gc_correct.0.err:

cd /home/some_directory/tmpBattenberg
/usr/bin/Rscript -e 'library(Battenberg); gc.correct.wgs(Tumour_LogR_file="87336744-255b-42a1-a1ff-0eded2655d8f_mutantLogR.tab", outfile="87336744-255b-42a1-a1ff-0eded2655d8f_mutantLogR_gcCorrected.tab", correlations_outfile="87336744-255b-42a1-a1ff-0eded2655d8f_GCwindowCorrelations.txt", gc_content_file_prefix="/home/xxxx/BATTENBERG/tmp2/reference_hg38/GC_correction_hg38/1000_genomes_GC_corr_chr_",replic_timing_file_prefix=NULL, chrom_names=as.vector(c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X"))) '
Warning message:
The path argument of write_tsv() is deprecated as of readr 1.4.0.
Please use the file argument instead.
This warning is displayed once every 8 hours.
Call lifecycle::last_warnings() to see where this warning was generated.
220.29user 28.44system 4:26.97elapsed 93%CPU (0avgtext+0avgdata 21400400maxresident)k
2455614inputs+10494816outputs (3major+3085355minor)pagefaults 0swaps

xxxxxxxxxxxxxxxxxx_imputefromaf.7.err:

cd /home/some_directory/result/sample2/hogehoge/tmpBattenberg
/usr/bin/Rscript -e 'library(Battenberg); generate.impute.input.wgs(chrom="chr7", tumour.allele.counts.file="87336744-255b-42a1-a1ff-0eded2655d8f_alleleFrequencies_chrchr7.txt", normal.allele.counts.file="dace3db7-74a8-408c-9872-d424ee7429bb_alleleFrequencies_chrchr7.txt", output.file="87336744-255b-42a1-a1ff-0eded2655d8f_impute_input_chrchr7.txt", imputeinfofile="/home/ny1fh/BATTENBERG/tmp2/reference_hg38/impute_info.txt", is.male="TRUE", problemLociFile="/home/ny1fh/BATTENBERG/tmp2/reference_hg38/probloci/probloci.txt.gz", useLociFile=NA, heterozygousFilter=0.1) '
Error in file(file, "rt") : invalid 'description' argument
Calls: generate.impute.input.wgs -> read.table -> file
0.26user 0.04system 0:00.30elapsed 99%CPU (0avgtext+0avgdata 73864maxresident)k
0inputs+48outputs (0major+18718minor)pagefaults 0swaps
####################################################################################

Apparently allelecount process completed with no error, and with complete set of xxxxxx_alleleFrequencies_chrZZZZ.txt.
Runbaflog.err file and gc.correct.err file seemed okay, but in the imputefromaf.err file I found the error above.

The 'chrom_names' variable first had no 'chr' prefix inside as you can see in the runbaflog.err file and gc.correct.file.
(chrom_names=as.vector(c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X"))
However, I also noticed 'chrom' variable in generate.impute.input.wgs function has 'chr'. and this function required 'xxxx_alleleFrequencies_chrchr7.txt' as inputfile. (in my result folder there's 'xxxx_alleleFrequencies_chr7.txt')

I believe the battenberg.R file has only one site defining chrom_names variable(line145-150) , and subsequent functions are called with each 'chrom' from 'chrom_names', so I wonder why chrom_names variable suddenly changed from without chr prefix to WITH chrom prefix.

Just so you know, the bam files are alinged with hg38, and has 'chr' prefix

I also tried recent versions of distributed source including dev branch with battenberg R package, as well as all previous images from quay:io, and they all got stuck in the getBAFsAndLogRs function, with empty xxxxx_mutantBAF.tab files (just as someone mentioned in the issue, even though I tried with more cores, more memory(eg, 128G+10core). So I gave these versions up.
ANY help would be appreciated..
thanks in advance!

**Comment by jcesar101** [comment]
Hi!,

as discussed in issue #114, this could be related to the reference files used, as they should be aligned with the input files having or not the "chr" prefix in the chromosome name. The subset of reference files with and without the prefix can be downloaded from [ORA](https://ora.ox.ac.uk/objects/uuid:08e24957-7e76-438a-bd38-66c48008cf52). Please try using the ones ending in "*_chr.*" and see if that solves this problem.

**Comment by hidehitofukushima** [comment]
I'll try these.thank you for quick reply!

**Comment by hidehitofukushima** [comment]
Hi. Just tried the reference file from ORA, and this time encountered similar, but slightly different issue.
I've unzipped all the files with "_chr" , renamed 1000G files and GC files like someone mentioned.

this time, the first error was in the 'runbaflog' err file.
I've checked how getBAFsAndLogRs is defined, and realised this require the ${TUMOR_NAME}_alleleFrequencies_chrchrN.txt, but my result only has ${TUMOR_NAME}_alleleFrequencies_chrN.txt file and ${TUMOR_NAME}_alleleFrequencies_chrN_splitY.txt....

> + cd /home/xxxx/BATTENBERG/result/sample/AG-AL-0008-T-01-D/tmpBattenberg
> + /usr/bin/Rscript -e 'library(Battenberg);  getBAFsAndLogRs(tumourAlleleCountsFile.prefix="87336744-255b-42a1-a1ff-0eded2655d8f_alleleFrequencies_chr", normalAlleleCountsFile.prefix="dace3db7-74a8-408c-9872-d424ee7429bb_alleleFrequencies_chr", figuresFile.prefix="87336744-255b-42a1-a1ff-0eded2655d8f_", BAFnormalFile="87336744-255b-42a1-a1ff-0eded2655d8f_normalBAF.tab", BAFmutantFile="87336744-255b-42a1-a1ff-0eded2655d8f_mutantBAF.tab", logRnormalFile="87336744-255b-42a1-a1ff-0eded2655d8f_normalLogR.tab", logRmutantFile="87336744-255b-42a1-a1ff-0eded2655d8f_mutantLogR.tab", combinedAlleleCountsFile="87336744-255b-42a1-a1ff-0eded2655d8f_alleleCounts.tab", chr_names=as.vector(c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chrX")), g1000file.prefix="/home/xxxx/BATTENBERG/reference_hg38/wtchr_wochr23/1000G_loci_hg38/1000genomesAlleles2012_chr", minCounts=10, samplename="87336744-255b-42a1-a1ff-0eded2655d8f", seed=1488823153) '
> Error in `[.data.frame`(allele_data, , 1) : undefined columns selected
> Calls: getBAFsAndLogRs -> gsub -> [ -> [.data.frame
> 0.36user 0.22system 0:03.91elapsed 15%CPU (0avgtext+0avgdata 73776maxresident)k
> 45420inputs+32outputs (76major+18642minor)pagefaults 0swaps

again facing  very similar 'chr' problem, Could you help me with this?
thanks, 
Hidehito

**Comment by jcesar101** [comment]
Hi!,

could you please try again with the following prefix values:

```
...
tumourAlleleCountsFile.prefix="87336744-255b-42a1-a1ff-0eded2655d8f_alleleFrequencies_", normalAlleleCountsFile.prefix="dace3db7-74a8-408c-9872-d424ee7429bb_alleleFrequencies_", 
...
```

**Comment by hidehitofukushima** [comment]
Hi, isn't the 'tumourAlleleCountsFile.prefix' designated in the source  already include 'chr' inside??  
prepare_wgs function has this part inside.... 
so should i alter the source code??

```
  # Obtain BAF and LogR from the raw allele counts
  getBAFsAndLogRs(tumourAlleleCountsFile.prefix=paste(tumourname,"_alleleFrequencies_chr", sep=""),
                  normalAlleleCountsFile.prefix=paste(normalname,"_alleleFrequencies_chr", sep=""),
                  figuresFile.prefix=paste(tumourname, "_", sep=''),
                  BAFnormalFile=paste(tumourname,"_normalBAF.tab", sep=""),
                  BAFmutantFile=paste(tumourname,"_mutantBAF.tab", sep=""),
                  logRnormalFile=paste(tumourname,"_normalLogR.tab", sep=""),
                  logRmutantFile=paste(tumourname,"_mutantLogR.tab", sep=""),
                  combinedAlleleCountsFile=paste(tumourname,"_alleleCounts.tab", sep=""),
                  chr_names=chrom_names,
                  g1000file.prefix=g1000allelesprefix,
                  minCounts=min_normal_depth,
                  samplename=tumourname)
  # Perform GC correction



```

**Comment by jcesar101** [comment]
Hi,

no, please don't change the code, I provided those instructions for the getBAFsAndLogRs function invocation of the previous message.

When running Battenberg, wether you have 1 or chr1 it should be consistent for all the sections were the chromosome name is used, so if you are getting this error is because either the reference files include the prefix but the input BAMs do not include it, or viceversa. Could you please share the log of the Battenberg pipeline execution to try to find what is causing this problem?

**Comment by toddajohnson** [comment]
Hi,
I was talking with Fukushima last night at a meeting and thought that I would take a look here about his problem.
Perhaps this is a naive comment, but the current version of Battenberg installed from GitHub using devtools::install_github("Wedge-Oxford/battenberg") would include a number of changes made in 2021 that seemed to address chromosome naming such as replacing "chr" in files using gsub.  However, Fukushima used the current Docker image from quay.io, but according to the update history of cancerit/cgbBattenberg for ver 3.7.1, that included changes up to October 21, 2020 (specifically, installed BB 2.2.10, so 10/20/2020), so it would be missing those chromosome name changes along with a number of others that, frankly, sound important when reading through the history.  Which version of BB should we be using?

**Comment by jcesar101** [comment]
Hi Todd,

that's correct the default GitHub installation already includes some additional functionality, but a formal release is still in progress. 

We are expecting to have the new release of Battenberg ready in a few months and, in the meantime, please keep using release 2.2.9 or 2.2.10, as they have been tested under a broader ser of conditions than the most up-to-date version in dev and master branches.

Thank you for using Battenberg and for your patience whilst the new release is completed.

**Comment by hidehitofukushima** [comment]
Dr Cortes, 
Sorry for the late reply.
Could you take a look at the log files? 
I 've uploaded the zip file of the working folder of battenberg along with the log files.
In there you cant find log files of later steps, because I've stopped the analysis halfway through, knowing some of the earlier steps were already with critical errors. And Ive kept only a few of allelefrequency files and 1000genome loci files and deleted the others because of size restriction.
[sample.zip](https://github.com/Wedge-lab/battenberg/files/14734216/sample.zip)

**Comment by jcesar101** [comment]
Hi,

thank you for sending more details about this execution.

Looks like the problem is due to the list of chromosome names received by the getAlleleCounts function not having the "chr" prefix:

```
      getAlleleCounts(bam.file=tumourbam,
                      output.file=paste(tumourname,"_alleleFrequencies_chr", chrom_names[i], ".txt", sep=""),
                      g1000.loci=paste(g1000prefix, chrom_names[i], ".txt", sep=""),
                      min.base.qual=min_base_qual,
                      min.map.qual=min_map_qual,
                      allelecounter.exe=allelecounter_exe)
```

as wether we work with or without this prefix, the most important thing is to be consistent across all input and reference files (please see issue #128 for a similar problem). In this case, looks like the list of chromosome names retrieved from the impute_info.txt does not contain the prefix (or at least not for all the chromosomes). Could you please check if that's the case and if there are any other reference (or input) files used in this experiment that don't include the "chr" prefix in the chromosome names?

**Comment by hidehitofukushima** [comment]
Dr Todd, 
Appreciate the analyses and the advice. While I'm still trying on this issue, I'm also back on native Battenberg code, I'll let you know when finally succeed. As you pointed out, 'chr'-or-without-'chr'-wise, there's a discrepancy between and the output filename of Allelecount process and  the expected names of input files later on. What I still don't understand is why chrom_names variable seemed 'changed' before and after the allelecount process. I'm explaining this below.


Dr Julio, 

I know precisely what you mean, so you're basically saying, in this case, output file in this getAlleleCounts funcition should be like,  '{TUMOR_NAME}_alleleFrequencies_chrchrN.txt' and so are the other filenames throughout the entire process.

The thing is, however, the impute_info.txt and all the other reference files I used DOES include 'chr' prefix. Even my bam files of course include 'chr' prefix and I think Dr Todd can confirm this.

Quite strange it seems, because the 'chrom_names' variable is defined only once in battenberg.R, and it should read impute_info.txt as dataframe and simply extract the chromosome column as strings. But as the log files show, at this point, the chrom_names obviously doesn't include 'chr' prefix.
However,   in the file I've shared with you, functions of later steps (e.g. GCcorrection) does include 'chr' prefix, which means chrom_names variable IS like c('chr1', 'chr2', 'chr3', 'chr4', ' chr5'...) at this point. 

 I still don't know why this happend, though I'm suspecting this could be related to splitting process before the allelecount process. Don't you have any suggestion regarding this?

**Comment by jcesar101** [comment]
Hi!,

it's strange indeed, and the only explanation I can think about at this point is that at least two different impute_info.txt files are being used for different (split) stages in the Battenberg pipeline. Are you using any of the R wrappers provided (inst/example folder) or do you have a custom script to execute the pipeline steps?

**Comment by fswirsky** [comment]
Hi,

I've come here from Query #114 as that was closed and I am still having the same runtime and "minCount=10" issue. I'm using hg19 samples and don't have the same set of reference files as hg38 so the "_chr.zip" files don't exist for me as recommended in that thread. I'm at a loss for what else to do any help is greatly appreciated.

Felix

**Comment by jcesar101** [comment]
Hi Felix,

that issue was closed because it involved different scenarios and different errors reported. For a better follow up, next time please try to raise a new issue each time a problem is detected.

Checking the messages from the other issue, I believe you are using Battenberg version 2.2.9 with R 4.3.1, is this correct?

Currently, we have tested versions 2.2.9 and 2.2.10 of Battenberg with R 4.1.2. Would it be possible to try this combination in your environment? Otherwise you might need to try the Battenberg version in the [dev](https://github.com/Wedge-lab/battenberg/tree/dev) branch with R 4.3.1, but that one is not fully tested yet. We are working on releasing a new version that supports newer version of R and other libraries, please bear with us whilst this new version is completed.


Kind regards and thank you.

**Comment by jcesar101** [comment]
This issue will be closed now as the original problem was not related with the versions of Battenberg maintained in this repository. For any other problems please raise a new issue.
